src/components/box/box.component.tsx

```diff
@@ -52,87 +57,70 @@ export interface BoxProps
   rowGap?: Gap;
   /** Design Token for Box Shadow. Note: please check that the box shadow design token you are using is compatible with the Box component. */
   boxShadow?: BoxShadowsType;
+  /** @private @ignore */
+  className?: string;
+  /** Set the color attribute of the Box component */
+  color?: string;
+  /** Set the bg attribute of the Box component */
+  bg?: string;
+  /** Set the backgroundColor attribute of the Box component */
+  backgroundColor?: string;
+  /** Set the opacity attribute of the Box component */
+  opacity?: string | number;
 }
 
-let isDeprecationWarningTriggered = false;
-
-export const Box = styled.div<BoxProps>`
-  ${() => {
-    if (!isDeprecationWarningTriggered) {
-      isDeprecationWarningTriggered = true;
-      Logger.deprecate(
-        "Previous props that could be spread to the `Box` component are being removed. Only props documented in the intefaces will be supported."
-      );
-    }
-    return css``;
-  }}
-
-  ${space}
-  ${layout}
-  ${flexbox}
-  ${position}
-
-  ${({ color, bg, backgroundColor, ...rest }) =>
-    styledColor({ color, bg, backgroundColor, ...rest })}
-
-  ${({ overflowWrap }) =>
-    overflowWrap &&
-    css`
-      overflow-wrap: ${overflowWrap};
-    `}
-
-  ${({ scrollVariant }) =>
-    scrollVariant &&
-    css`
-      scrollbar-color: ${boxConfig[scrollVariant].thumb}
-        ${boxConfig[scrollVariant].track};
-
-      &::-webkit-scrollbar {
-        width: 8px;
-      }
-      &::-webkit-scrollbar-track {
-        background-color: ${boxConfig[scrollVariant].track};
-      }
-      &::-webkit-scrollbar-thumb {
-        background-color: ${boxConfig[scrollVariant].thumb};
-      }
-    `}
-
-  ${({ boxSizing }) =>
-    boxSizing &&
-    css`
-      box-sizing: ${boxSizing};
-    `}
-
-    ${({ display, gap, columnGap, rowGap }) =>
-    (display === "flex" || display === "inline-flex") &&
-    css`
-      ${gap !== undefined &&
-      css`
-        gap: ${boxConfig.gap(gap)};
-      `}
-
-      ${columnGap !== undefined &&
-      css`
-        column-gap: ${boxConfig.gap(columnGap)};
-      `}
-
-      ${rowGap !== undefined &&
-      css`
-        row-gap: ${boxConfig.gap(rowGap)};
-      `}
-    `};
-
-  ${({ boxShadow }) =>
-    boxShadow &&
-    css`
-      box-shadow: var(--${boxShadow});
-    `}
-`;
-
-Box.defaultProps = {
-  theme: BaseTheme,
-};
+export const Box = React.forwardRef<HTMLDivElement, BoxProps>(
```

Comment:
Hardcoding the `HTMLDivElement` is improper due to `as` support - it will vary with the parameter - just like it previously did with styled-components. This causes issues with `onClick` not being available despite setting `as="button"` - while at the same time the `handleKeyPress` is working fine ⚠️ 

<img width="746" alt="Zrzut ekranu 2023-01-25 o 14 51 33" src="https://user-images.githubusercontent.com/30114244/214582938-35b0c713-66db-4f90-9fbe-54e779ca8241.png">

<img width="746" alt="Zrzut ekranu 2023-01-25 o 14 51 44" src="https://user-images.githubusercontent.com/30114244/214582968-543086a1-5e98-43c4-97cf-c499b2ceabef.png">

<img width="746" alt="Zrzut ekranu 2023-01-25 o 14 53 19" src="https://user-images.githubusercontent.com/30114244/214583015-9df4638d-93ea-4523-880c-3c546d511f3f.png">

<img width="230" alt="Zrzut ekranu 2023-01-25 o 15 00 23" src="https://user-images.githubusercontent.com/30114244/214583190-c4ac280b-597c-479a-8d72-a5cdf950556d.png">

This is kind of a blocker for us (SBS-49786).